datafusion/core/tests/dataframe/mod.rs

```diff
@@ -5673,3 +5673,43 @@ async fn test_fill_null_all_columns() -> Result<()> {
     assert_batches_sorted_eq!(expected, &results);
     Ok(())
 }
+
+#[tokio::test]
+async fn test_sql_support_sql_to_view_types() -> Result<()> {
```

Comment:
Instead of adding rust based tests, could you please make these sqllogictest based instead? sqllogictest is easier to maintain and run

The instructions are here
https://github.com/apache/datafusion/tree/main/datafusion/sqllogictest

Perhaps as a new test in https://github.com/apache/datafusion/blob/main/datafusion/sqllogictest/test_files/ddl.slt